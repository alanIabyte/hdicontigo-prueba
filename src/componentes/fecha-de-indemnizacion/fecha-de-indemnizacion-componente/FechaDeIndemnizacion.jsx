import React from "react";
import PropTypes from "prop-types";
import {
  EnvolvedorFechaDeIndemnizacion,
  FechaDeIndemnizacionCampo,
  FechaDeIndemnizacionValor,
} from "./FechaDeIndemnizacion.styled";
import obtenerValorDeArregloDeStrings from "../../utilidades-compartidas/obtenerValorDeArregloDeStrings";
import agregarFormatoDeFecha from "../../utilidades-compartidas/agregarFormatoFecha";

const diccionario = {
  fechaDeIndemnizacion: "Pago programado para el día:",
};

const FechaDeIndemnizacion = (props) => {
  const { datosFechaDeIndemnizacion } = props;

  const fechaDeIndemnizacion = obtenerValorDeArregloDeStrings(
    datosFechaDeIndemnizacion,
    "FechaIndemnización: "
  );

  const fechaDeIndemnizacionFormato =
    agregarFormatoDeFecha(fechaDeIndemnizacion);

  return (
    <EnvolvedorFechaDeIndemnizacion>
      <FechaDeIndemnizacionCampo>
        {diccionario.fechaDeIndemnizacion}
      </FechaDeIndemnizacionCampo>
      <FechaDeIndemnizacionValor>
        {fechaDeIndemnizacionFormato}
      </FechaDeIndemnizacionValor>
    </EnvolvedorFechaDeIndemnizacion>
  );
};

FechaDeIndemnizacion.defaultProps = {
  datosFechaDeIndemnizacion: ["Fechadeindemnizacion:  "],
};

FechaDeIndemnizacion.propTypes = {
  datosFechaDeIndemnizacion: PropTypes.arrayOf(PropTypes.string),
};

export default FechaDeIndemnizacion;
