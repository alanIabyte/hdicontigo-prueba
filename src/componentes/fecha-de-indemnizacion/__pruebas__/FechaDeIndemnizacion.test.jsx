import React from "react";
import renderer from "react-test-renderer";
import FechaDeIndemnizacion from "../fecha-de-indemnizacion-componente/FechaDeIndemnizacion";

describe("FechaDeIndemnizacion", () => {
  it("se despliega correctamente", () => {
    const tree = renderer.create(<FechaDeIndemnizacion />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
