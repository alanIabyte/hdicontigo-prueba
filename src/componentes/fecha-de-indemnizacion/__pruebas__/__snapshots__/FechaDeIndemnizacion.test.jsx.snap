// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FechaDeIndemnizacion se despliega correctamente 1`] = `
<div
  className="sc-ikkxIA"
>
  <div
    className="sc-kpDqfm sc-dAbbOL bTCkFB hdZOgq"
  >
    Pago programado para el día:
  </div>
  <div
    className="sc-kpDqfm sc-feUZmu bTCkFB jMhLTa"
  >
    Fecha inválida
  </div>
</div>
`;
